src/coprocessor/readpool_impl.rs

```diff
@@ -132,3 +146,25 @@ pub fn tls_collect_read_flow(region_id: u64, statistics: &crate::storage::Statis
         flow_stats.add(&statistics.data.flow_stats);
     });
 }
+
+#[inline]
+pub fn with_tls_engine_any<F, R>(f: F) -> R
```

Comment:
Oh, why not saving an Engine Box directly? I don't like using `any` mechanism.

Reply:
The `Engine` trait cannot be used as dyn trait, because it must be implemented with `Clone` trait which makes it not `Sized`.
So I use `any` mechanism directly, I can't find a better way to achieve this......